### ISSUE TITLE
visual.textStim not displaying text centered

### ISSUE THREAD

**Issue by themeo** [psychopy/psychopy#2691]
I'm running PsychoPy 3.2.4 on Windows 10. 

When running simple 
```
text = visual.TextStim(win=win, name='text',
    text='Some text\nSome more text\n',
    font='Arial',
    pos=(0, 0), height=0.1, wrapWidth=None, ori=0, 
    color='white', colorSpace='rgb', opacity=1, 
    languageStyle='LTR',
    depth=0.0, alignHoriz='center');
```

the text is horizontally left-aligned instead of being centered. I'm using 'norm' units, a single laptop monitor. The results are the same for full-screen mode and window mode. There are no error messages.

**Comment by peircej** [comment]
Yes. Actually this is an issue with the underlying pyglet text object, which doesn't support alignment. All this achieves is alignment of the *box* according to the x_pos. We're working on our own text rednering engine that will have more flexible formating including text alignment. I'm closing this for now as there isn't an immediate fix and it will be fixedd by TextBox2

**Comment by themeo** [comment]
Is there _any_ way in which the alignment of the text or the textbox can be manipulated then? Is there any other solution even if it is not too elegant? 

Besides, it stands in clear contradiction to what the manuals say so I think this still remains an issue -- even if there are no immediate fixes (I was surprised by your closing the ticket).

**Comment by peircej** [comment]
I closed the ticket because it's already being developed. In fact it's done, but we're just polishing the interface to it. This is coming very soon (in the next release, which will be out in the next couple of weeks).
No, right now the only way to center-align things is to break them into lines and center each one.

**Comment by themeo** [comment]
That is great news, thanks for the answer. However, my current problem is that I cannot center-align any single line not containing any newline characters:

```
text = visual.TextStim(win, text='+++', alignHoriz='center')
```

... displays the text left-aligned.

**Comment by peircej** [comment]
Could you tell me how you've installed psychopy, and particularly pyglet? I wonder if you're using the version from github (and/or pyglet 1.4)?

When I run the following code in PsychoPy *Standalone* (which comes with pyglet 1.3) I get the text in the middle of the screen as expected. When I run it in the github master branch I get left-aligned **if I use pyglet 1.4** as noted in #2590 

``` python
from psychopy import visual, event

win = visual.Window()
text = visual.TextStim(win, text='+++', alignHoriz='center')
text.draw()
win.flip()
event.waitKeys()
```

When I run it in this branch 
  https://github.com/peircej/psychopy/tree/textPyglet1.4
with pyglet 1.4 I can then adapt the critical line to be
``` python
text = visual.TextStim(win, text='+++', anchorHoriz='center')
```
and all is good again. (In the new version I'm updating `alignHoriz` to be a pair of values: `alignText` and `anchorHoriz` to make explicit the different jobs they do

**Comment by peircej** [comment]
That new branch is where I've been fixing things to work with pyglet1.4 and, along the way, providing access to full centered align for multiline text

**Comment by themeo** [comment]
I used the master branch from github (3.2.4) and pyglet 1.4.8. Standalone
Python was not working for some reason (the starting script closed
immediately after launching it).

After downgrading pyglet to 1.3.2 I'm able to center single lines without a
problem.
I have just switched to PsychoPy and I'm starting a new experiment next
week, so I was eager to find a solution ASAP.
Thanks for your help and looking forward to PsychoPy being able to fully
cooperate with pyglet 1.4+!
